tls/s2n_security_policies.c

```diff
@@ -960,6 +996,10 @@ struct s2n_security_policy_selection security_policy_selection[] = {
     { .version = "PQ-TLS-1-2-2023-04-09", .security_policy = &security_policy_pq_tls_1_2_2023_04_09, .ecc_extension_required = 0, .pq_kem_extension_required = 0 },
     { .version = "PQ-TLS-1-2-2023-04-10", .security_policy = &security_policy_pq_tls_1_2_2023_04_10, .ecc_extension_required = 0, .pq_kem_extension_required = 0 },
     { .version = "PQ-TLS-1-3-2023-06-01", .security_policy = &security_policy_pq_tls_1_3_2023_06_01, .ecc_extension_required = 0, .pq_kem_extension_required = 0 },
+    { .version = "PQ-TLS-1-3-2023-10-07", .security_policy = &security_policy_pq_tls_1_3_2023_10_07, .ecc_extension_required = 0, .pq_kem_extension_required = 0 },
```

Comment:
nit: I think the TLS version in the policy name typically indicates the minimum TLS version supported, not the maximum. It might be more consistent to have the policies be named PQ-TLS-1-2-*.

Reply:
This policy technically supports TLS 1.2, but not for PQ -- only for compatibility. Clients who try to initiate a TLS 1.2 connection will negotiate classical ECDH-only. So, I guess the question is whether the version should indicate the minimum PQ TLS version (1.3) or the minimum overall TLS version (1.2). I don't have strong opinions either way.

What do you folks think @goatgoose @alexw91 ?

Reply:
I agree with @goatgoose, the policy names should start with `PQ-TLS-1-2-*`

Reply:
Sounds good, I'll update the names.